### PR TITLE
Watch the manifest file for changes

### LINKED ISSUE #87: ViteManifest does not reload if the manifest file changes on a running site
**Describe the bug**
If you have a running ASP.NET Core site and you compile and deploy a new build of the client files _only_ (no new binaries so the site doesn't need to restart), `ViteManifest` will still have the old manifest loaded in memory and attempt to serve files that may no longer exist (because the hash changed, for example). This causes any affected page to no longer render due to missing scripts.

**To Reproduce**
1. Start your site, but without the dev server (`vite build`)
2. Navigate to your page, see that it works
3. Change your client file in any way and rerun `vite build`. Presumably, you have different hashes of the file now 
4. Refresh your page, see that it tries to load the old JavaScript files.

**Expected behavior**
`Vite.AspNetCore` should watch the manifest file for changes. If the file is updated, it should invalidate the cached data and reload.

**Device (please complete the following information):**
 - OS: Windows
 - IDE/Editor: Visual studio 2022, but not sure that is Relevant here.

**Additional context**
Note that, while I didn't actually try this out yet, it is possible that a workaround could be to implement your own `IViteManifest` wrapper around `ViteManifest` and add this behavior. Still, I would imagine it should ultimately be built into this library since I don't think there is a reason you wouldn't want this behavior.

The [FileProvider](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.fileproviders.physicalfileprovider?view=dotnet-plat-ext-8.0) may be of interest here.

Might be interested in taking the PR if I have a chance here and you agree with the proposed change. But feel free to take it if you feel like you can knock it out quickly.

### PR DESCRIPTION
**Pull request by someguy20336**
Fixes #87 

In the PR, I am watching for changes in the manifest so `ViteManifest` is refreshed when the file is updated.  This should ensure that upon deploying new client files, we aren't attempting to serve files from the old manifest.  It also avoids needing to change the scope of the service (resulting in reading that file for every request).

I also considered using the [Configuration Providers](https://learn.microsoft.com/en-us/dotnet/core/extensions/configuration-providers#json-configuration-provider) but ultimately didn't think that saved _that much_ effort - but let me know that you think.

I was able to hack together a unit test, but let me know if you think of a better way to do it.

Resources:
- [File Providers](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/file-providers?view=aspnetcore-8.0#watch-for-changes)

### CODE REVIEW

File: src/ViteNetProject.Tests/ManifestTest.cs
Comment (Eptagone) [inline src/ViteNetProject.Tests/ManifestTest.cs:1]: The "\\" only works on Windows. The tests fails on Linux/MacOS.

**Review by Eptagone — CHANGES_REQUESTED** [review]
It works, thanks! Just please review the `//` error in the tests and also, add the missing nullable declarations in `ViteManifest.cs` file.
```log
Vite.AspNetCore\src\Vite.AspNetCore\Services\ViteManifest.cs(36,9): warning C
S8618: Non-nullable field 'chunks' must contain a non-null value when exiting constructor. Consider declari
ng the field as nullable. [C:\Users\Quetzal\source\repos\Vite.AspNetCore\src\Vite.AspNetCore\Vite.AspNetCor
e.csproj]
Vite.AspNetCore\src\Vite.AspNetCore\Services\ViteManifest.cs(36,9): warning C
S8618: Non-nullable field 'changeToken' must contain a non-null value when exiting constructor. Consider de 
claring the field as nullable. [C:\Users\Quetzal\source\repos\Vite.AspNetCore\src\Vite.AspNetCore\Vite.AspN 
etCore.csproj]
Vite.AspNetCore\src\Vite.AspNetCore\Services\ViteManifest.cs(36,9): warning C 
S8618: Non-nullable field 'changeTokenDispose' must contain a non-null value when exiting constructor. Cons 
ider declaring the field as nullable. [C:\Users\Quetzal\source\repos\Vite.AspNetCore\src\Vite.AspNetCore\Vi 
te.AspNetCore.csproj]
Vite.AspNetCore\src\Vite.AspNetCore\Services\ViteManifest.cs(36,9): warning C 
S8618: Non-nullable field 'viteOptions' must contain a non-null value when exiting constructor. Consider de 
claring the field as nullable. [C:\Users\Quetzal\source\repos\Vite.AspNetCore\src\Vite.AspNetCore\Vite.AspN 
etCore.csproj]
Vite.AspNetCore\Services\ViteManifest.cs(36,9): warning C 
S8618: Non-nullable field 'fileProvider' must contain a non-null value when exiting constructor. Consider d 
eclaring the field as nullable. [C:\Users\Quetzal\source\repos\Vite.AspNetCore\src\Vite.AspNetCore\Vite.Asp 
NetCore.csproj]
```

File: src/ViteNetProject.Tests/ManifestTest.cs
Comment (someguy20336) [reply src/ViteNetProject.Tests/ManifestTest.cs:1]: ah, of course I didn't think about this.  Thanks!

**Comment by someguy20336** [comment]
Should be all good on the warnings.  `chunks` was still guaranteed to be set, but I don't think it is smart enough to follow the code into `InitializeManifest`, so I used the `null!` syntax there and made a comment.  Let me know if you prefer something else.

**Comment by Eptagone** [comment]
Looks good for me. I'm merging now

**Pull request merged**